src/domain/collaboration/collaboration/collaboration.service.ts

```diff
@@ -363,4 +364,51 @@ export class CollaborationService {
       collaborationID
     );
   }
+
+  public async updateCalloutsSortOrder(
+    collaboration: ICollaboration,
+    sortOrderData: UpdateCollaborationCalloutsSortOrderInput
+  ): Promise<ICallout[]> {
+    const collaborationLoaded = await this.getCollaborationOrFail(
+      collaboration.id,
+      {
+        relations: ['callouts'],
+      }
+    );
+    const allCallouts = collaborationLoaded.callouts;
+    if (!allCallouts)
+      throw new EntityNotFoundException(
+        `Callout not initialised, no Callouts: ${collaboration.id}`,
+        LogContext.COLLABORATION
+      );
+
+    // Get the callouts specified
+    const calloutsInOrder: ICallout[] = [];
+    let minCalloutSortOrder = 10000;
+    for (const calloutID of sortOrderData.calloutIDs) {
+      let callout;
+      if (calloutID.length === UUID_LENGTH)
+        callout = allCallouts.find(callout => callout.id === calloutID);
+      else callout = allCallouts.find(callout => callout.nameID === calloutID);
+
+      if (!callout) {
+        throw new EntityNotFoundException(
+          `Callout with requested ID (${calloutID}) not located within current Collaboration: ${collaboration.id}`,
+          LogContext.COLLABORATION
+        );
+      }
+      if (callout.sortOrder < minCalloutSortOrder) {
+        minCalloutSortOrder = callout.sortOrder;
+      }
+      calloutsInOrder.push(callout);
+    }
+
+    // Now update all the callouts sort order
+    for (const callout of calloutsInOrder) {
+      callout.sortOrder = minCalloutSortOrder;
+      minCalloutSortOrder += 2;
```

Comment:
Should be 1?

Reply:
I thought 2 gave a bit of space; does not really matter as long as they go up...